ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java

```diff
@@ -714,23 +714,27 @@ private final void shutdown(boolean waitForTransactions)
         synchronized (this) {
             _inShutdown = true;
 
-            /*
-                * If the caller does not want to wait for the normal transaction timeout
-                * periods to elapse before terminating, then we first start by enabling
-                * our time machine!
-                */
-
-            if (!waitForTransactions) {
-                _reaperElements.setAllTimeoutsToZero();
-            }
-
             /*
                 * Wait for all of the transactions to
                 * terminate normally.
                 */
             while (!_reaperElements.isEmpty()) {
+                /*
+                 * If the caller does not want to wait for the normal transaction timeout
+                 * periods to elapse before terminating, then we first start by enabling
+                 * our time machine!
+                 */
+
+                if (!waitForTransactions) {
+                    _reaperElements.setAllTimeoutsToZero();
+                    if (_dynamic) {
+                        nextDynamicCheckTime.set(0);
```

Comment:
Please can I ask the intention by only setting this when _dynamic is true? For example it is [used](https://github.com/jbosstm/narayana/blob/main/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java#L97) in the check method outside even when the reaper is not dynamic?

Reply:
@tomjenkinson my reading of https://github.com/jbosstm/narayana/blob/64b02cc09db97498219bef1269cf98eed4eba309/ArjunaCore/arjuna/classes/com/arjuna/ats/arjuna/coordinator/TransactionReaper.java#L571 and the checkPeriod method made the intention of nextDynamicCheckTime seem to only apply in the dynamic case. 

But as you say there is inconsistent usage of it in check/reinsertElement, and removeElementReaper. So it's probably best to omit the _dynamic check here as well.

Reply:
Changed to remove the _dynamic check. A follow-up issue should be considered on making its usage more consistent.